src/foam/nanos/rope/ROPEUtils.js

```diff
@@ -0,0 +1,167 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.rope',
+  name: 'NextRelationshipsList',
+  
+  documentation: `
+  This is the List of next relationships used in the sub-maps of CRUDMap and in relationshipMap.
+  The reason this model is needed is because when a ROPE with the maps as Map<String, List<String>> is read from the journal, 
+  the List is interpreted as an String[] instead of List<String> (probably), so we get <String, Ljava.lang.String>.
+  Modelling this list, and using this class as the value type in the maps is a quick fix to this problem.
+  `,
+
+  properties: [
+    {
+      name: 'nextRelationships',
+      class: 'List',
+      javaType: 'java.util.List<String>'
+    }
+  ]
+
+});
+
+foam.CLASS({
+  package: 'foam.nanos.rope',
+  name: 'RelationshipMap',
+
+  properties: [
+    {
+      name: 'map',
+      class: 'Map',
+      javaType: 'java.util.Map<String, NextRelationshipsList>'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'get',
+      javaType: 'java.util.List<String>',
+      args: [
+        { name: 'relationshipKey', class: 'String' }
+      ],
+      javaCode: `
+        return getMap().containsKey(relationshipKey) && getMap().get(relationshipKey) != null ? getMap().get(relationshipKey).getNextRelationships() : null;
+      `
+    },
+  ]
+
+});
+
+foam.CLASS({
+  package: 'foam.nanos.rope',
+  name: 'CRUDMap',
+
+  javaImports: [
+    'java.util.Map',
+    'java.util.List'
+  ],
+
+  constants: [
+    {
+      name: 'DEFAULT',
+      type: 'String',
+      value: '__default__',
+      documentation: `
+      This is the default key to be used for mapping to a list of next relationships in the sub-maps of this model.
+      To authorize using a different set of relationships for create/update operations, please use propertyname as key.
+      `
+    },
+  ],
+
+  properties: [
+    {
+      name: 'create',
+      class: 'Map',
+      javaType: 'java.util.Map<String, NextRelationshipsList>'
+    },
+    {
+      name: 'read',
+      class: 'Map',
+      javaType: 'java.util.Map<String, NextRelationshipsList>'
+    },
+    {
+      name: 'update',
+      class: 'Map',
+      javaType: 'java.util.Map<String, NextRelationshipsList>',
+    },
+    {
+      name: 'delete',
+      class: 'Map',
+      javaType: 'java.util.Map<String, NextRelationshipsList>',
+    }
+  ],
+  
+  axioms: [
+    {
+      name: 'javaExtras',
```

Comment:
Could these be modelled as FOAM methods rather than a javaExtras axiom? Then documentation properties can be added to them, and other future FOAMy things will be compatible.

Reply:
Oops, missed the comment